### DISCUSSION TITLE
Why does my Placement alternate not work?

### DISCUSSION THREAD

**Discussion by benschi11** [OrchardCMS/OrchardCore#12166] [Q&A]
I have an ContentType "Fluid Page" with a FlowPart attached. I want to render a alternate shape for the form widget if a specific url is requested (in my case "/kontakt"). I tried this with the follwing placement file named "FlowPart":
```
[
  { 
    "place": null, 
    "displayType": "Detail", 
    "differentiator": null,
    "alternates": [
      "FlowPart__ContactForm"
    ],
    "wrappers": null,
    "shape": null,
    "path": [
      "/kontakt"
    ],
    "contentType": [
      "FlowPart"
    ]
  }
]
```

Then I created in my Theme a "FlowPart-ContactForm.cshtml". The debug output from the Lombiq module displays following:

``` 
FlowPart
Alternates: FlowPart_Detail, Form__FlowPart, Widget__FlowPart, Form_Detail__FlowPart, Widget_Detail__FlowPart
Binding sources: Areas/OrchardCore.Flows/Views/FlowPart.cshtml, Areas/Hausleitner.Theme/Views/FlowPart-ContactForm.cshtml
Differentiator: FlowPart
DisplayType: Detail
IsCached: False
Name: FlowPart
Position: 5
Prefix: FlowPart
```

I find my Shape in the Binding source, but not listed first - is that the problem? How can I change it?
Thanks a lot!

**Comment by ns8482e** [comment]
Did you create placement.json?

**Reply by benschi11** [reply]
I went in the admin: "Design - Placements" and created the placement from the first post.

**Reply by ns8482e** [reply]
Can you check if that works as placement.json as file?

**Reply by benschi11** [reply]
Thanks for the answer, but unfortunately it does not work with the placement file in the Theme either. The funny part is, that it shows in the debug output from the Lombiq Module under "Binding Source".

**Reply by ns8482e** [reply]
Bindingsource tells what’s view that could possibly run for FlowPart,  but it only applies when alternate exists- the last in list will apply if matched with binding source and here  alternate is not on that list

**Reply by ns8482e** [reply]
I guess one of the rule you defined  failing to apply the placement
